internal/resources/iampolicy/resource_iam_policy_test.go

```diff
@@ -0,0 +1,598 @@
+	namespaceresource "github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/namespace"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/policy"
+	scoperesource "github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/policy/scope"
+	testhelper "github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/testing"
+	workspaceresource "github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/workspace"
+)
+
+const (
+	IAMPolicyResource          = ResourceName
+	IAMPolicyResourceVar1      = "test_iam_policy_1"
+	IAMPolicyResourceVar2      = "test_iam_policy_2"
+	clusterResource            = clusterresource.ResourceName
+	clusterResourceVar         = "test_cluster"
+	clusterGroupResource       = clustergroupresource.ResourceName
+	clusterGroupResourceVar    = "test_cluster_group"
+	workspaceResource          = workspaceresource.ResourceName
+	workspaceResourceVar       = "test_workspace"
+	namespaceResource          = namespaceresource.ResourceName
+	namespaceResourceVar       = "test_namespace"
+	managementClusterName      = scoperesource.AttachedValue
+	provisionerName            = scoperesource.AttachedValue
+	clusterName                = "tf-attach-test"
+	workspaceName              = "default"
+	namespaceName              = "tf-namespace"
+	clusterGroupNameForCluster = "default"
+	clusterGroupNamePrefix     = "tf-cg-test"
+	subject1Name               = "test-1"
+	subject1Kind               = "GROUP"
+	orgRole                    = "organization.view"
+	clusterGroupRole           = "clustergroup.view"
+	clusterRole                = "cluster.view"
+	workspaceRole              = "workspace.view"
+	namespaceRole              = "namespace.view"
+)
+
+type testAcceptanceConfig struct {
+	Provider                   *schema.Provider
+	ClusterResource            string
+	ClusterResourceVar         string
+	ClusterResourceName        string
+	ClusterGroupResourceName   string
+	KubeConfigPath             string
+	ClusterName                string
+	ClusterRole1               string
+	ClusterGroupNameForCluster string
+	Meta                       string
+	ManagementClusterName      string
+	ProvisionerName            string
+	ClusterGroupResource       string
+	ClusterGroupResourceVar    string
+	ClusterGroupName           string
+	ClusterGroupRole1          string
+	OrgID                      string
+	OrgRole1                   string
+	WorkspaceResourceName      string
+	WorkspaceResource          string
+	WorkspaceResourceVar       string
+	WorkspaceName              string
+	WorkspaceRole1             string
+	NamespaceResourceName      string
+	NamespaceResource          string
+	NamespaceResourceVar       string
+	NamespaceName              string
+	NamespaceRole1             string
+	IAMPolicyResource          string
+	IAMPolicyResourceVar1      string
```

Comment:
can you have them as an array ? 
and so many values, are they needed ? can you group them ?